snapcraft/tests/integration/__init__.py

```diff
@@ -196,6 +197,25 @@ def copy_project_to_cwd(self, project_dir: str) -> None:
             os.path.join(self.snaps_dir, project_dir), self.path,
             preserve_symlinks=True)
 
+    def construct_yaml(self, name='test', version='0.1',
```

Comment:
This is better than it was before, but I feel like this is introducing yet another inconsistency in how we're doing this. I'd still rather see snaps in integration/snaps/ that we can test standalone as well, like the other grammar tests. However, I've already made that argument, so I won't block on it.